modules/siddhi-extensions/kf/src/main/java/org/wso2/siddhi/extension/kf/Kalman2DFilterWithMatrices.java

```diff
@@ -0,0 +1,239 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.kf;
+
+import org.apache.commons.math3.linear.LUDecomposition;
+import org.apache.commons.math3.linear.MatrixUtils;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.exception.ExecutionPlanRuntimeException;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.query.processor.stream.function.StreamFunctionProcessor;
+import org.wso2.siddhi.query.api.definition.AbstractDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+public class Kalman2DFilterWithMatrices extends StreamFunctionProcessor {
+
+    private ConcurrentHashMap<String, ConcurrentHashMap<Integer, RecordLocatorKalmonFilter>> kalmonFilterHashMap;
+
+    @Override
+    public void start() {
+
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    @Override
+    public Object[] currentState() {
+        return new Object[0];
+    }
+
+    @Override
+    public void restoreState(Object[] state) {
+
+    }
+
+    @Override
+    protected Object[] process(Object[] data) {
+        if (data[0] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to geo:kalman2DFilter() " +
+                                                    "function. First argument cannot be null");
+        }
+        if (data[1] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to geo:kalman2DFilter() " +
+                                                    "function. First argument cannot be null");
+        }
+        if (data[2] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to geo:kalman2DFilter() " +
+                                                    "function. First argument cannot be null");
+        }
+        if (data[3] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to geo:kalman2DFilter() " +
+                                                    "function. First argument cannot be null");
+        }
+        if (data[4] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to geo:kalman2DFilter() " +
+                                                    "function. First argument cannot be null");
+        }
+
+        double measuredValue = (Double) data[0];
+        double measuredValue2 = (Double) data[1];
+        double measuredChangingRate = 0.003;
+        long timestamp = (Long) data[2];
+        String recordLocator = (String) data[3];
+        int floor = (Integer) data[4];
```

Comment:
Can do that.

And also "measuredChangingRate" i have set it to 0.003 since its human movement. I will make it a function parameter as well so it can be changed. wdyt?